dali/pipeline/operator/eager_operator.h

```diff
@@ -169,28 +170,28 @@ class DLL_PUBLIC EagerOperator {
                        ", instance name: \"", name_, "\", encountered:\n", what);
   }
 
-  static inline std::shared_ptr<ThreadPool> GetSharedThreadPool() {
-    std::lock_guard lock(shared_thread_pool_mutex_);
-
-    if (!shared_thread_pool_) {
-      int num_cores = std::thread::hardware_concurrency();
-      num_cores = num_cores < 6 ? num_cores : 6;
+  static inline int GetDefaultNumThreads() {
+    int num_cores = std::thread::hardware_concurrency();
+    return num_cores < 6 ? num_cores : 6;
+  }
 
-      shared_thread_pool_ =
-          std::make_shared<ThreadPool>(num_cores, CPU_ONLY_DEVICE_ID, false, "EagerOperator");
-    }
+  static inline std::shared_ptr<ThreadPool> &CreateSharedThreadPool() {
```

Comment:
I'd rename this to `GetSharedThreadPool`. Otherwise it looks as if we are creating it on every call.